# Check definitions

In [1]:
# 1. What does interest rate do to a peroson's default probablity?
#2. Diff of amt_current and amt_total current
# 3. amount recerivable , total receivable, priciple recerivable
# 4. name_contract status

In [110]:
import pandas as pd
import numpy as np

PATH = '/home/kai/data/kaggle/homecredit/'
train = pd.read_csv(PATH + 'application_train.csv')
print('train')
test = pd.read_csv(PATH + 'application_test.csv')
print('test')
bureau = pd.read_csv(PATH + 'bureau.csv')
print('bureau')
# previous_application = pd.read_csv(PATH + 'previous_application.csv')
# print('previous')
# bureau_balance = pd.read_csv(PATH + 'bureau_balance.csv')
# print('bureau_balance')
installments_payments = pd.read_csv(PATH + 'installments_payments.csv')
print('installments_payments done')
credit_card_balance = pd.read_csv(PATH + 'credit_card_balance.csv')
print('credit_card_balance')
POS_CASH_balance = pd.read_csv(PATH + 'POS_CASH_balance.csv')
print('POS_CASH_balance')
# credit_card_balance = pd.read_csv(PATH + 'credit_card_balance.csv')
# print('credit_card_balance')
previous_application = pd.read_csv(PATH + 'previous_application.csv')
print('previous_application')

train
test
bureau
installments_payments done
credit_card_balance
POS_CASH_balance
previous_application


In [3]:
cred = credit_card_balance.copy()
print('shape of credit', cred.shape)

shape of credit (3840312, 23)


# Analysis of the percentage of SK_ID_CURR into train& test

In [104]:
set_cred_id = set(cred['SK_ID_CURR'].values)
set_train_id = set(train['SK_ID_CURR'].values)
set_test_id = set(test['SK_ID_CURR'].values)
print('cred into train', (1 - len(set_train_id - set_cred_id)/len(set_train_id)))
print('cred into test', (1 - len(set_test_id - set_cred_id)/len(set_test_id)))

cred into train 0.2826077766323807
cred into test 0.34164204825209255


# AMT_RECEIVABLE, AMT_TOTAL_RECEIVABLE, AMT_RECEIVABLE_TOTAL

In [15]:
list = [col for col in cred.columns.values if 'REC' in col]

In [16]:
list

['AMT_RECEIVABLE_PRINCIPAL', 'AMT_RECIVABLE', 'AMT_TOTAL_RECEIVABLE']

In [17]:
cred_rec = cred[list]
cred_rec.isnull().sum()/len(cred_rec)

AMT_RECEIVABLE_PRINCIPAL    0.0
AMT_RECIVABLE               0.0
AMT_TOTAL_RECEIVABLE        0.0
dtype: float64

In [18]:
(cred_rec['AMT_RECEIVABLE_PRINCIPAL']>cred_rec['AMT_RECIVABLE']).sum()

107994

In [43]:
(cred_rec['AMT_RECEIVABLE_PRINCIPAL']<=cred_rec['AMT_RECIVABLE']).sum()

3732318

In [44]:
(cred_rec['AMT_RECEIVABLE_PRINCIPAL']!=cred_rec['AMT_RECIVABLE']).sum()

1685587

In [20]:
(cred_rec['AMT_TOTAL_RECEIVABLE']>cred_rec['AMT_RECIVABLE']).sum()

11244

In [21]:
(cred_rec['AMT_TOTAL_RECEIVABLE']<=cred_rec['AMT_RECIVABLE']).sum()

3829068

In [37]:
df = cred_rec[cred_rec['AMT_RECEIVABLE_PRINCIPAL']==cred_rec['AMT_RECIVABLE']]
df.shape

(2154725, 3)

In [42]:
_df = df[df['AMT_RECEIVABLE_PRINCIPAL'] ==df['AMT_RECEIVABLE_PRINCIPAL']]
_df.shape

(2154725, 3)

In [48]:
prev = previous_application.copy()
status_prev = [col for col in prev.columns if 'STATUS' in col]

In [52]:
status_prev

['NAME_CONTRACT_STATUS']

In [53]:
set(prev.NAME_CONTRACT_STATUS)

{'Approved', 'Canceled', 'Refused', 'Unused offer'}

In [56]:
x = set(cred.NAME_CONTRACT_STATUS)

In [58]:
for i in x:
    print(i)
    print((cred['NAME_CONTRACT_STATUS'] == i).sum()/len(cred))

Completed
0.03356966829778414
Active
0.9630561266897064
Demand
0.000355439870510521
Signed
0.0028794535443995174
Approved
1.3019775476575862e-06
Refused
4.426723662035793e-06
Sent proposal
0.00013358289638966834


In [59]:
cred.isnull().sum()/len(cred)

SK_ID_PREV                    0.000000
SK_ID_CURR                    0.000000
MONTHS_BALANCE                0.000000
AMT_BALANCE                   0.000000
AMT_CREDIT_LIMIT_ACTUAL       0.000000
AMT_DRAWINGS_ATM_CURRENT      0.195249
AMT_DRAWINGS_CURRENT          0.000000
AMT_DRAWINGS_OTHER_CURRENT    0.195249
AMT_DRAWINGS_POS_CURRENT      0.195249
AMT_INST_MIN_REGULARITY       0.079482
AMT_PAYMENT_CURRENT           0.199981
AMT_PAYMENT_TOTAL_CURRENT     0.000000
AMT_RECEIVABLE_PRINCIPAL      0.000000
AMT_RECIVABLE                 0.000000
AMT_TOTAL_RECEIVABLE          0.000000
CNT_DRAWINGS_ATM_CURRENT      0.195249
CNT_DRAWINGS_CURRENT          0.000000
CNT_DRAWINGS_OTHER_CURRENT    0.195249
CNT_DRAWINGS_POS_CURRENT      0.195249
CNT_INSTALMENT_MATURE_CUM     0.079482
NAME_CONTRACT_STATUS          0.000000
SK_DPD                        0.000000
SK_DPD_DEF                    0.000000
dtype: float64

In [61]:
(cred['AMT_BALANCE']<=cred['AMT_CREDIT_LIMIT_ACTUAL']).sum()

3497958

In [65]:
1 ==np.nan

False

In [66]:
(cred['AMT_BALANCE']>=cred['AMT_INST_MIN_REGULARITY']).sum()/len(cred)

0.8798430960817767

In [67]:
(cred['AMT_BALANCE']<=cred['AMT_CREDIT_LIMIT_ACTUAL']).sum()/len(cred)

0.9108525557298469

In [69]:
x = (cred['AMT_INST_MIN_REGULARITY']<=cred['AMT_CREDIT_LIMIT_ACTUAL']).sum()/len(cred)

In [70]:
0.079482+x

0.9993445007551469

In [72]:
(cred['AMT_DRAWINGS_CURRENT']<cred['AMT_CREDIT_LIMIT_ACTUAL']).sum()/len(cred)

0.7979260539247853

In [85]:
(cred['AMT_PAYMENT_CURRENT']>=cred['AMT_PAYMENT_TOTAL_CURRENT']).sum()/len(cred)

0.8000193734259091

In [78]:
(cred['AMT_PAYMENT_CURRENT']>cred['AMT_PAYMENT_TOTAL_CURRENT']).sum()/len(cred)

0.30914988157212225

In [82]:
(cred['AMT_RECIVABLE']==cred['AMT_CREDIT_LIMIT_ACTUAL']).sum()/len(cred)

0.18274192305208536

In [91]:
(cred['AMT_PAYMENT_CURRENT']<=cred['AMT_RECIVABLE']).sum()/len(cred)

0.4557926543468343

In [93]:
(cred['AMT_RECIVABLE']<=cred['AMT_BALANCE']).sum()/len(cred)

0.9725399915423538

In [95]:
# AMT_DRAWINGS_CURRENT
((cred['AMT_DRAWINGS_CURRENT']+cred['AMT_RECIVABLE'])<=cred['AMT_BALANCE']).sum()/len(cred)

0.8231372867621173

In [96]:
((cred['AMT_DRAWINGS_CURRENT']+cred['AMT_RECIVABLE'])<=cred['AMT_PAYMENT_CURRENT']).sum()/len(cred)

0.39768565679038576

In [ ]:
(cred['AMT_RECIVABLE']<=cred['AMT_BALANCE']).sum()/len(cred)

In [98]:
set(cred['SK_DPD_DEF'])

{0,
 1,
 5,
 6,
 7,
 8,
 9,
 10,
 11,
 12,
 16,
 17,
 18,
 19,
 26,
 27,
 28,
 29,
 30,
 31,
 32,
 38,
 39,
 42,
 46,
 47,
 48,
 49,
 56,
 57,
 58,
 59,
 60,
 61,
 62,
 63,
 68,
 69,
 77,
 79,
 80,
 89,
 90,
 91,
 92,
 93,
 99,
 107,
 110,
 120,
 121,
 122,
 123,
 124,
 130,
 138,
 141,
 151,
 152,
 153,
 154,
 161,
 169,
 172,
 179,
 182,
 183,
 184,
 185,
 189,
 199,
 200,
 213,
 214,
 215,
 216,
 230,
 243,
 244,
 245,
 246,
 260,
 274,
 275,
 276,
 277,
 291,
 304,
 305,
 306,
 307,
 322,
 335,
 336,
 337,
 338,
 352,
 366,
 367,
 383,
 395,
 396,
 397,
 398,
 413,
 426,
 427,
 428,
 444,
 456,
 457,
 458,
 459,
 475,
 487,
 488,
 489,
 490,
 503,
 517,
 518,
 519,
 520,
 547,
 548,
 549,
 550,
 578,
 579,
 580,
 581,
 608,
 609,
 610,
 611,
 612,
 639,
 640,
 641,
 669,
 670,
 671,
 672,
 700,
 701,
 702,
 703,
 731,
 732,
 760,
 761,
 762,
 763,
 791,
 792,
 793,
 821,
 822,
 823,
 824,
 852,
 853,
 854,
 855,
 882,
 883,
 884,
 885,
 913,
 914,
 915,
 916,
 943,
 944,
 945,
 946

In [101]:
(cred['SK_DPD'] ==1).sum()/len(cred)

0.02353168180085368

In [102]:
(cred['SK_DPD_DEF'] == 0).sum()/len(cred)

0.9767362651784542

In [112]:
# install = installments_payments.copy()
pos = POS_CASH_balance
prev = previous_application.copy()

set_cred_id = set(cred['SK_ID_CURR'].values)
set_train_id = set(train['SK_ID_CURR'].values)
set_test_id = set(test['SK_ID_CURR'].values)
print('cred into train', (1 - len(set_train_id - set_cred_id)/len(set_train_id)))
print('cred into test', (1 - len(set_test_id - set_cred_id)/len(set_test_id)))

cred into train 0.2826077766323807
cred into test 0.34164204825209255


'''
cred:
cred into train 0.2826077766323807
cred into test 0.34164204825209255
install:
cred into train 0.9483985938714388
cred into test 0.9835877236172658
POS:
cred into train 0.9412476301660754
cred into test 0.9807976366322009
prev:
cred into train 0.9464929709831518
cred into test 0.9806335138683735
bureau:
cred into train 0.8568506492450676
cred into test 0.8682094206466437
'''